The --version-script flag is not recognized by all linkers so just remove it.

--- meson.build.orig
+++ meson.build
@@ -207,7 +206,6 @@
 vflag = '-Wl,--version-script,@0@/@1@'.format(meson.current_source_dir(), mapfile)
 
 lib = library('esmtp', sources,
-	      link_args : vflag, link_depends : mapfile,
 	      soversion : libesmtp_so_version,
 	      dependencies : deps,
 	      install : true)
